### PR TITLE
feat(perf): org browser retrieve perf enhancements

### PR DESCRIPTION
**Pull request by rcoringrato-sfdc**
### What does this PR do?

- Utilize source deploy/retrieve library for Org Browser retrieve operations

### What issues does this PR fix or reference?
@W-8242567@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts [hunk 1]
```diff
@@ -132,16 +140,92 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
     }
   }
 }
+export class LibraryRetrieveSourcePathExecutor extends LibraryCommandletExecutor<
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts:143]: Do we have a test to cover this new code ?

Comment (rcoringrato-sfdc) [reply packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts:143]: @lcampos No, I'll add test cases for it.

Comment (lcampos) [reply packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts:143]: Ok, everything else looks good btw

File: packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts [hunk 2]
```diff
@@ -59,11 +62,11 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
 
     execution.processExitSubject.subscribe(async exitCode => {
       let properties;
-      const registryAccess = new RegistryAccess();
       try {
-        const components = registryAccess.getComponentsFromPath(
-          execFilePathOrPaths
-        );
+        const ws = new WorkingSet();
+        const comps = ws.resolveSourceComponents(execFilePathOrPaths);
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts:67]:
```suggestion
        const ws = WorkingSet.fromSource(execFilePathOrPaths);
```

Shorthand for initializing WorkingSet with source

File: packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts [hunk 3]
```diff
@@ -132,16 +140,92 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
...
+
+    const conn = await workspaceContext.getConnection();
+    const result = await ws.retrieve(conn, output, { merge: true });
+
+    if (result.success && this.openAfterRetrieve) {
+      const compSet = ws.resolveSourceComponents(output);
+      await this.openResources(this.findResources(components[0], compSet));
+    }
+
+    return result.success;
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts:178]: Do we need to output the result table to the channelService?

**Comment by brpowell** [comment]
Looks good aside from the one comment I had about the channel service. I discovered a conversion bug that's part of the library through these changes and filed an issue with it.

**Review by brpowell — APPROVED** [review]

**Pull request merged**